### PR TITLE
closes #286 commit messaging, new r-cmd vignette

### LINKED ISSUE #286: Documentation:  Close issues when merged into `devel`
### Please select a category the issue is focused on?

Developer Guides

### Let us know where something needs a refresh or put your idea here!

Our current strategy is to close issues only when merged into `main`.  This was to keep issues open in case users identified the same issues multiple times and we could point to why we are closing the duplicate issue.  As this hasn't really been a problem for us - I think we can safely abandon this strategy.

**New Strategy:** Close issues when merging into `devel` branch

**Benefits**:  
- Backlog is kept to a minimum.  
- Easier for Backlog Triage team to review issues.
- More focus can be put on larger issues as they remain in the backlog page 1 rather than getting pushed onto a new page.

Definition of Done

- [ ] Programming Strategy is updated to reflect this new process
- [ ] Communicated to team on this new process
- [ ] Default branches changed to `devel`

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiraldev/devel/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
HI all,

Some minor updates on commit messaging on the vignettes.  I also split on the R CMD section of the Pull Request Vignette into its own vignette.  It was buried at the end there.

Did a little other sprucing throughout to bring up to alignment with our current practices.

File: vignettes/git_usage.Rmd [hunk 1]
```diff
@@ -82,26 +81,48 @@ You can also make use of the Git Tab within RStudio to commit your changes. A be
 knitr::include_graphics("github_committ.png", dpi = 144)
 ```
 
+### Commit Message Etiquette 
+
+We require developers to **insert the issue number** into each commit message. Placing the issue number in your commit message allows reviewers to quickly find discussion surrounding your issue. When pushed to GitHub the issue number will be hyperlink to the issue tracker. A powerful tool for discussion and traceability, which we think is valuable in a highly regulated industry like Pharma.
```
Comment (zdz2101) [inline vignettes/git_usage.Rmd:86]:
```suggestion
We require developers to **insert the issue number** into each commit message. Placing the issue number in your commit message allows reviewers to quickly find discussion surrounding your issue. When pushed to GitHub the issue number will be hyperlinked to the issue tracker, a powerful tool for discussion and traceability, which we think is valuable in a highly regulated industry like Pharma.
```

Basic grammar tweak

**Review by zdz2101 — COMMENTED** [review]
Do we need to add a NEWS snippet at least to indicate we created a new section?

**Comment by bms63** [comment]
> Do we need to add a NEWS snippet at least to indicate we created a new section?

Sure - I love the news!

**Review by zdz2101 — APPROVED** [review]

**Pull request merged**